core/src/test/scala/org/apache/spark/rdd/PipedRDDSuite.scala

```diff
@@ -50,6 +50,29 @@ class PipedRDDSuite extends SparkFunSuite with SharedSparkContext {
     }
   }
 
+  test("failure in iterating over pipe input") {
+    if (testCommandAvailable("cat")) {
+      val nums =
+        sc.makeRDD(Array(1, 2, 3, 4), 2)
+          .mapPartitionsWithIndex((index, iterator) => {
+            new Iterator[Int] {
+              def hasNext = true
+              def next() = {
+                throw new SparkException("Exception to simulate bad scenario")
+              }
+            }
+          })
+
+      val piped = nums.pipe(Seq("cat"))
+
+      intercept[SparkException] {
+        piped.collect()
+      }
+    } else {
+      assert(true)
```

Comment:
This can be removed?

Also in the test above, we need to assert that an exception was thrown, not just not-fail if it was thrown. Otherwise it passes if no exception is throw from `collect()`?